bigtable/admin.go

```diff
@@ -1129,13 +1185,44 @@ func (iac *InstanceAdminClient) DeleteCluster(ctx context.Context, instanceID, c
 	return err
 }
 
-// UpdateCluster updates attributes of a cluster
+// SetAutoscaling enables autoscaling on a cluster. To remove autoscaling, use
+// UpdateCluster.
+func (iac *InstanceAdminClient) SetAutoscaling(ctx context.Context, instanceID, clusterID string, conf AutoscalingConfig) error {
+	ctx = mergeOutgoingMetadata(ctx, iac.md)
+	cluster := &btapb.Cluster{
+		Name:   "projects/" + iac.project + "/instances/" + instanceID + "/clusters/" + clusterID,
+		Config: conf.proto(),
+	}
+	lro, err := iac.iClient.PartialUpdateCluster(ctx, &btapb.PartialUpdateClusterRequest{
+		UpdateMask: &field_mask.FieldMask{
+			Paths: []string{"cluster_config.cluster_autoscaling_config"},
+		},
+		Cluster: cluster,
+	})
+	if err != nil {
+		return err
+	}
+	return longrunning.InternalNewOperation(iac.lroClient, lro).Wait(ctx, nil)
+}
+
+// UpdateCluster updates attributes of a cluster. If Autoscaling is configured
```

Comment:
Unfortunately, this method only changes the number of serve nodes. I'll update the documentation to make that clear. Since we can't change this function, we're effectively stuck with the name. As an alternative to this approach here (SetAutoscaling turns it on, UpdateCluster turns it off), we could introduce a more generic function, e.g., UpdateClusterConfig or similar.

Reply:
I think introducing a more general UpdateClusterConfig (or similar) is a good idea, but has its own gotcha's. So I propose we punt on that work for now, especially since such a future change could simply refactor the internals of SetAutoscaling to use the more general update method.